### PR TITLE
Milagro BLS update

### PR DESCRIPTION
**Pull request by kirk-baird**
## Issue Addressed

N/a

## Proposed Changes

- Update function naming in `milagro_bls`
- Tidy up `SignatureSets`
- Increase handling of references in `milagro_bls` to reduce copying
- Add helper functions for `Signature` -> `AggregateSignature` and `PublicKey` to `AggregatePublicKey`

## Additional Info

-  Wait until merging of https://github.com/sigp/milagro_bls/pull/24 and a release is made then update lighthouse to point to the release
- `SecretKey` now must be less than the curve group order (which is a little less than 2^256) . Thus `SecretKey` from bytes may fail if more than 31 bytes have been set, and will fail if more than 32 bytes are set.

### CODE REVIEW

**Comment by kirk-baird** [comment]
The `milagro_bls` updates have been made into a release and links updated here. This can now be merged after review :)

File: beacon_node/beacon_chain/src/beacon_chain.rs [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             |validator_index| {
                 pubkey_cache
                     .get(validator_index)
-                    .map(|pk| Cow::Borrowed(pk.as_point()))
+                    .map(|pk| Cow::Borrowed(pk))
```
[the remark this answers is not in the document]
Comment (kirk-baird) [reply beacon_node/beacon_chain/src/beacon_chain.rs]: Done :)

File: beacon_node/beacon_chain/src/beacon_chain.rs [hunk 2]
```diff
@@ -1412,7 +1412,7 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                     if validator_index < state.validators.len() {
                         validator_pubkey_cache
                             .get(validator_index)
-                            .map(|pk| Cow::Borrowed(pk.as_point()))
+                            .map(|pk| Cow::Borrowed(pk))
```
[the remark this answers is not in the document]
Comment (kirk-baird) [reply beacon_node/beacon_chain/src/beacon_chain.rs]: Done :)

**Comment by kirk-baird** [comment]
I've merged in master and fixed the conflicts so now it should be able to merge with master.

**Comment by paulhauner** [comment]
Re-running the jobs cause the network test failed it :)

**Comment by paulhauner** [comment]
Oh no, this didn't get merged in time and has conflicts. I can come and fix them up, it's probably due to my changes.

I'd also like to double-check this is going to mesh will the the existing two v0.11.x testnets we're running.

**Comment by kirk-baird** [comment]
> Oh no, this didn't get merged in time and has conflicts. I can come and fix them up, it's probably due to my changes.

:( hopefully soon!

> I'd also like to double-check this is going to mesh will the the existing two v0.11.x testnets we're running.

Yep it's v0.11.x friendly!

**Comment by paulhauner** [comment]
> :( hopefully soon!

I promise this isn't going to linger! I just need to address #983 then I can come do this :)

**Comment by paulhauner** [comment]
I just fixed the conflicts! I'll merge it in once it passes CI :)

**Pull request merged**